### PR TITLE
Fix array params and  removing unnecessary question marks

### PR DESCRIPTION
**Pull request by hasanaktas**
## Status

**READY/WIP/HOLD**

## Description

Array Params Example

color:['red','blue','green']
?colors=red&color='blue'&color='green'

QuestionMark Example

params={}
old getUrl Result
/product/12?

new getUrl Result
/product/12

## Related PRs

List related PRs against other branches:

| branch              | PR       |
| ------------------- | -------- |
| other_pr_production | [link]() |
| other_pr_master     | [link]() |

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Changelog Entry (unreleased)

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.

```bash
> git pull --prune
> git checkout <branch>
> grunt jasmine
```

1.

### CODE REVIEW

**Comment by soartec-lab** [comment]
@hasanaktas 
Thanks. I would like to do this, so would it be possible for you to share the minimum schema and orval.config to reproduce?

File: packages/fetch/src/index.ts [hunk 1]
```diff
@@ -49,16 +49,27 @@ ${
...
+      });
     }
+
+    return normalizedParams.append(key, value.toString());
   });`
     : ''
 }
 
-  return \`${route}${queryParams ? '?${normalizedParams.toString()}' : ''}\`
+  return \`${route}${queryParams && normalizedParams.size ? '?${normalizedParams.toString()}' : ''}\`
```
Comment (soartec-lab) [inline packages/fetch/src/index.ts:72]: this dosn't work we hopeful so i fixed syntax error :)

```suggestion
  ${queryParams
    ? `return normalizedParams.size ? \`${route}${'?${normalizedParams.toString()}'}\` : \`${route}\``
    : `return \`${route}\``
  }
```

and generated bellow:

```ts
return normalizedParams.size ? `http://localhost:8000/pets?${normalizedParams.toString()}` : `http://localhost:8000/pets`
```

File: packages/fetch/src/index.ts [hunk 2]
```diff
@@ -49,16 +49,27 @@ ${
   const normalizedParams = new URLSearchParams();
 
   Object.entries(params || {}).forEach(([key, value]) => {
+    if (value === undefined) {
```
Comment (soartec-lab) [inline packages/fetch/src/index.ts:52]: i don't think this is necessary, so could you tell me why you added it ?

**Review by soartec-lab — COMMENTED** [review]
@han

First, i reviewed that remove question mark in query params. i'll check array params later.

**Comment by melloware** [comment]
@soartec-lab where are we at with this PR? Should it be closed?

**Comment by soartec-lab** [comment]
@melloware 

No, they are right and we need this PR.

@hasanaktas 

Do you plan on continuing this?
If there is no response, I will respond.

**Comment by melloware** [comment]
@soartec-lab i am going to close this since you have taken it over.

**Comment by soartec-lab** [comment]
@melloware thanks!

**Pull request closed without merging**